src/jiraUpdateTicket.gs

```diff
@@ -17,70 +18,77 @@ function updateJiraIssues(headerRow, dataRows) {
   debug.info('updateJiraIssues called Keys:--%s  DATA ROWS %s', Object.keys(headerRow).join(","), dataRows);
   var result = { rowsUpdated: 0, status: false, message: "", finished: false, errors: [] };
 
-  if (headerRow === null || Object.keys(headerRow).length == 0) {
-    result.finished = true;
-    result.message = "No values in header row";
+  if (hasValidationErrors()) {
     return result;
   }
 
-  if (dataRows == null || dataRows.length == 0) {
-    result.status = false;
+  var allJiraFields = getAllJiraFields();
+  if (allJiraFields.length == 0) {
     result.finished = true;
-    result.message = "No issues were selected from your sheet";
+    result.status = false;
+    result.message = "Could not retrieve all field definitions from JIRA";
     return result;
-
   }
 
   var statusTransitioner = new IssueTransitioner();
-  allJiraFields = getAllJiraFields(
```

Comment:
Instead of using the call back I returned the value - makes the code easier to read imho